src/modules/direnv.rs

```diff
@@ -0,0 +1,302 @@
+use std::borrow::Cow;
+use std::fmt::Display;
+use std::path::PathBuf;
+use std::str::FromStr;
+
+use super::{Context, Module, ModuleConfig};
+
+use crate::configs::direnv::DirenvConfig;
+use crate::formatter::StringFormatter;
+
+/// Creates a module with the current direnv rc
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let mut module = context.new_module("direnv");
+    let config = DirenvConfig::try_load(module.config);
+
+    let direnv_applies = !config.disabled
+        && context
+            .try_begin_scan()?
+            .set_files(&config.detect_files)
+            .is_match();
+
+    if !direnv_applies {
+        return None;
+    }
+
+    let direnv_status = &context.exec_cmd("direnv", &["status"])?.stdout;
+    let state = match DirenvState::from_str(direnv_status) {
+        Ok(s) => s,
+        Err(e) => {
+            log_error(e);
+
+            return None;
+        }
+    };
+
+    let parsed = StringFormatter::new(config.format).and_then(|formatter| {
+        formatter
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "rc_path" => Some(Ok(state.rc_path.to_string_lossy())),
+                "allowed" => Some(Ok(Cow::from(state.allowed.to_string()))),
+                "loaded" => state
+                    .loaded
+                    .then_some("loaded")
+                    .or(Some("not loaded"))
+                    .map(Cow::from)
+                    .map(Ok),
```

Comment:
I think it should be possible to customize the different `allowed`/`loaded`-strings (or symbols).